### ISSUE TITLE
lens on GHC HEAD runs afoul new default type signature check

### ISSUE THREAD

**Issue by ezyang** [ekmett/lens#712]
Building with GHC HEAD as of commit 639e702b6129f501c539b158b982ed8489e3d09c, I get this error building lens:

```
src/Control/Lens/Indexed.hs:508:11: error:
    • The default type signature for itraverse:
        forall (f :: * -> *) a b.
        Applicative f =>
        (Int -> a -> f b) -> t a -> f (t b)
      does not match its corresponding non-default type signature
    • When checking the class method:
        itraverse :: forall i (t :: * -> *).
                     TraversableWithIndex i t =>
                     forall (f :: * -> *) a b.
                     Applicative f =>
                     (i -> a -> f b) -> t a -> f (t b)
      In the class declaration for ‘TraversableWithIndex’
```

This has affected other packages as well, but it is generally a bug in the package, not GHC. See: http://ghc.haskell.org/trac/ghc/ticket/13258 and http://ghc.haskell.org/trac/ghc/ticket/13249

Tagging #709.

**Comment by ezyang** [comment]
Weirdly, the error went away after I tried to rebuild. Closing I guess...

**Comment by glguy** [comment]
This was resolved recently in #711
